Test failure because of umask

The umask for my user under Linux is set to 0027.  When running `go run build.go test` locally with that umask, the test suite fails:

    --- FAIL: TestChmodDir (0.00s)
        basicfs_test.go:126: wrong perm: true 0750
    FAIL
    FAIL	github.com/syncthing/syncthing/lib/fs	0.408s

This is caused by the `Mkdir()` function subtracting the effective umask from the given mode argument, [as documented](https://golang.org/pkg/os/#Mkdir).  So the test can only succeed if the developer's umask allows creation of directories with a mode of 755.

I'm not sure whether this has any real-world effects besides the test suite failing.  Might just be an overly cautious test case.

### Version Information

Syncthing Version: v1.5.0-rc.1
Development environment: Linux Ubuntu 18.04 LTS

--- a/lib/fs/basicfs_test.go
+++ b/lib/fs/basicfs_test.go
@@ -121,6 +121,10 @@ func TestChmodDir(t *testing.T) {
 	if err := os.Mkdir(path, mode); err != nil {
 		t.Error(err)
 	}
+	// On UNIX, Mkdir will subtract the umask, so force desired mode explicitly
+	if err := os.Chmod(path, mode); err != nil {
+		t.Error(err)
+	}
 
 	if stat, err := os.Stat(path); err != nil || stat.Mode()&os.ModePerm != mode {
 		t.Errorf("wrong perm: %t %#o", err == nil, stat.Mode()&os.ModePerm)

